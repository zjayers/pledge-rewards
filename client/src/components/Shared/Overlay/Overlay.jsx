import React from 'react';
import Overlay from './Overlay.style';

export default (props) => {
  const { activated, setActivated } = props;

  const handleOverlayClick = () => {
    setActivated(!activated);
  };

  if (!activated) {
    return (
      <Overlay onClick={handleOverlayClick}>
        <p>Select this reward</p>
      </Overlay>
    );
  }

  /* Return a Fragment */
  return <></>;
};
