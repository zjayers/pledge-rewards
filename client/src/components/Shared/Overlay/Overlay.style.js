import styled from 'styled-components';

export default styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100%;
  width: 100%;
  opacity: 0;
  transition: 0.5s ease;
  background-color: ${(props) => props.theme.colors.highlightGreen};
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  z-index: 999;

  & p {
    font-size: 1.8rem;
    color: #ffffff;
  }

  &:hover {
    opacity: 0.9;

    & p {
      animation-name: focus-in-contract;
      animation-duration: 0.3s;
      animation-timing-function: ease;
      animation-delay: 0s;
      animation-iteration-count: 1;
      animation-direction: normal;
      animation-fill-mode: none;
    }
  }

  @keyframes focus-in-contract {
    0% {
      letter-spacing: 1em;
      filter: blur(6px);
      opacity: 0;
    }
    100% {
      filter: blur(0);
      opacity: 1;
    }
  }
`;
