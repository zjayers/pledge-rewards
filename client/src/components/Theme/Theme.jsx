import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  colors: {
    borderGrey: '#9B9E9E',
    highlightGreen: '#009E74',
    buttonGreen: '#009E74',
    gradientBlue: '#DBE7FE',
    gradientOrange: '#FFF2EC',
    text: '#282828',
    flavorText: '#888B8B',
    linkText: '#498F80',
  },
  fonts: ['Muli', 'sans-serif'],
  fontSizes: {
    small: '1rem',
    medium: '1.8rem',
    large: '3rem',
  },
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
