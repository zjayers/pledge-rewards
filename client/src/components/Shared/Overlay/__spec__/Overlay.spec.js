import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Overlay from '../Overlay';
import Theme from '../../../Theme/Theme';

describe('Overlay Component', () => {
  let component;
  const mockFn = jest.fn();
  beforeEach(() => {
    component = mount(
      <Theme>
        <Overlay activated={false} setActivated={mockFn} />
      </Theme>
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should return am empty fragment if the "activated" prop is true', () => {
    component.unmount();
    component = shallow(<Overlay activated />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should close the modal when description button is clicked', () => {
    component.simulate('click', () => {});
    expect(mockFn).toHaveBeenCalled();
  });
});
