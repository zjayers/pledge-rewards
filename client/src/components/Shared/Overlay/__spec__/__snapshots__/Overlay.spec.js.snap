// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Component should match the test snapshot 1`] = `
<Theme>
  <ThemeProvider
    theme={
      Object {
        "colors": Object {
          "borderGrey": "#9B9E9E",
          "buttonGreen": "#009E74",
          "flavorText": "#888B8B",
          "gradientBlue": "#DBE7FE",
          "gradientOrange": "#FFF2EC",
          "highlightGreen": "#009E74",
          "linkText": "#498F80",
          "text": "#282828",
        },
        "fontSizes": Object {
          "large": "3rem",
          "medium": "1.8rem",
          "small": "1rem",
        },
        "fonts": Array [
          "Muli",
          "sans-serif",
        ],
      }
    }
  >
    <_default
      activated={false}
      setActivated={[MockFunction]}
    >
      <Overlaystyle___default
        onClick={[Function]}
      >
        <div
          className="Overlaystyle___default-trsgfk-0 fGHlti"
          onClick={[Function]}
        >
          <p>
            Select this reward
          </p>
        </div>
      </Overlaystyle___default>
    </_default>
  </ThemeProvider>
</Theme>
`;

exports[`Overlay Component should return am empty fragment if the "activated" prop is true 1`] = `<Fragment />`;
